cocotb/regression.py

```diff
@@ -226,6 +228,32 @@ def _discover_tests() -> Iterable[Test]:
             )
             raise AttributeError("Test(s) %s doesn't exist in %s" % (tests, modules))
 
+    @classmethod
+    def _setup_pytest_assertion_rewriting(cls, test_modules: Iterable[str]) -> None:
+        try:
+            import pytest
+        except ImportError:
+            _logger.info(
+                "pytest not found, install it to enable better AssertionError messages"
+            )
+        else:
+            try:
+                # Install the assertion rewriting hook, which must be done before we
+                # import the test modules.
+                from _pytest.assertion import install_importhook
+                from _pytest.config import Config
+
+                test_modules_str = " ".join(f"{name}.py" for name in test_modules)
+                pytest_conf = Config.fromdictargs(
+                    {}, ["--capture=no", "-o", f"python_files={test_modules_str!r}"]
+                )
```

Comment:
I would guess that this works just as well as
```suggestion
                pytest_conf = Config.fromdictargs(
                    dict(python_files=["{name}.py" for name in test_modules],
                    ["--capture=no"]
                )
```
Does it?

Reply:
Unfortunately it does not. https://github.com/pytest-dev/pytest/discussions/10052#discussioncomment-3026738